### PR TITLE
refactored tag-resources function into class resource-tagger and fixe…

### PR DESCRIPTION
**Pull request by rafvasq**
…d tests; renamed files

### CODE REVIEW

File: lib/resource-tagger.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+  constructor({runaws, ec2, keyPrefix}) {
+    assert(typeof runaws === 'function');
+    this.runaws = runaws;
+
+    assert(typeof ec2 === 'object');
+    this.ec2 = ec2;    
+ 
+    assert(typeof keyPrefix === 'string');
+    assert(keyPrefix[keyPrefix.length - 1] === ':');
```
Comment (auni53) [inline lib/resource-tagger.js]: Should group type-checking asserts at the top, then constructing logic underneath

File: test/cloud-watch-event-listener_test.js [hunk 2]
```diff
@@ -51,17 +52,19 @@ describe('Cloud Watch Event Listener', () => {
...
 
-    listener = new CloudWatchEventListener({state, sqs, ec2, region, monitor, keyPrefix: cfg.app.keyPrefix});
+    listener = new CloudWatchEventListener({state, sqs, ec2, region, monitor, keyPrefix: cfg.app.keyPrefix, tagger});
   });
 
   // I could add these helper functions to the actual state.js class but I'd
   // rather not have that be so easy to call by mistake in real code
   beforeEach(async () => {
     await state._runScript('clear-db.sql');
+    sandbox.stub(tagger, 'runaws');
```
Comment (SamanthaYu) [inline test/cloud-watch-event-listener_test.js]: What is sandbox.stub() used for?

File: test/cloud-watch-event-listener_test.js [hunk 3]
```diff
@@ -51,17 +52,19 @@ describe('Cloud Watch Event Listener', () => {
 
     let monitor = await main('monitor', {profile: 'test', process: 'test'});
     let cfg = await main('cfg', {profile: 'test', process: 'test'});
+    tagger = await main('tagger', {profile: 'test', process: 'test'});
 
     await state._runScript('drop-db.sql');
     await state._runScript('create-db.sql');
 
-    listener = new CloudWatchEventListener({state, sqs, ec2, region, monitor, keyPrefix: cfg.app.keyPrefix});
+    listener = new CloudWatchEventListener({state, sqs, ec2, region, monitor, keyPrefix: cfg.app.keyPrefix, tagger});
```
Comment (auni53) [inline test/cloud-watch-event-listener_test.js]: Seems weird to mix both object literal forms in one line; can you add `const keyPrefix = cfg.app.keyPrefix` above this?

File: lib/cloud-watch-event-listener.js [hunk 4]
```diff
@@ -67,6 +67,9 @@ class CloudWatchEventListener extends events.EventEmitter {
     // Store the reference to the monitor instance
     this.monitor = monitor.prefix('cloud-watch-events');
 
+    assert(typeof tagger === 'object');
+    this.tagger = tagger;
```
[the remark this answers is not in the document]
Comment (rstiyer) [reply lib/cloud-watch-event-listener.js]: We are storing it because we use the tagger later in the _handler function

File: lib/api.js [hunk 5]
```diff
@@ -157,6 +157,7 @@ api.declare({
     try {
       await this.tagger.tagResources({
         keyPrefix: this.keyPrefix,
```
Comment (jhford) [inline lib/api.js]: We still don't need to pass in the keyprefix, but otherwise looks great!

**Comment by jhford** [comment]
This was merged with https://github.com/taskcluster/ec2-manager/commit/124f438aa2f4f0d5e7f09e6b939ad34bfd572eda

File: test/spot-request-poll_test.js [hunk 6]
```diff
@@ -33,131 +32,183 @@ describe('Spot Request Polling', () => {
...
+        return Promise.reject(Error('Line should not be reached'));
+      } catch(e) { }
+    });
+
+    it('fails with list containing non-string regions', () => {
+      try {
+        const poller = new SpotRequestPoller({ec2: {}, regions: [defaultSR.region, 1], state, runaws: () => {}});
+        return Promise.reject(Error('Line should not be reached'));
+      } catch(e) { }
+    });
```
Comment (walac) [inline test/spot-request-poll_test.js]: I think [`assert.throws`](https://nodejs.org/api/assert.html#assert_assert_throws_block_error_message) is the way here.

File: test/spot-request-poll_test.js [hunk 7]
```diff
@@ -33,131 +32,183 @@ describe('Spot Request Polling', () => {
...
-    assume(mock.callCount).equals(2);
-    assume(mock.firstCall.args[1]).equals('describeSpotInstanceRequests');
-    assume(mock.secondCall.args[1]).equals('cancelSpotInstanceRequests');
-    assume(mock.secondCall.args[2]).deeply.equals({
-      SpotInstanceRequestIds: [defaultSR.id],
+  describe('_poll helper method', () => {
+
+    it('succeeds with a valid region', async () => {
+      const poller = new SpotRequestPoller({ec2: {}, regions: [defaultSR.region], state, runaws: () => {}});
+      await poller._poll('foobar');
```
Comment (walac) [inline test/spot-request-poll_test.js]: Doesn't the call produce any side effects to check?

**Pull request closed without merging**